go/vt/vtgate/planbuilder/route.go

```diff
@@ -517,19 +517,8 @@ func (rb *route) isLocal(col *sqlparser.ColName) bool {
 // This will be used on the RHS node to fetch field info if the LHS
 // returns no result.
 func (rb *route) generateFieldQuery(sel *sqlparser.Select, jt *jointab) string {
-	formatter := func(buf *sqlparser.TrackedBuffer, node sqlparser.SQLNode) {
+	fallthroughHandler := func(buf *sqlparser.TrackedBuffer, node sqlparser.SQLNode) {
```

Comment:
I think this part can be simplified. How about something like this instead:
```
formatter := func(buf *sqlparser.TrackedBuffer, node sqlparser.SQLNode) {
  switch node := node.(type) {
  case *sqlparser.ColName:
    // like before
  case * sqlparser.TableName:
    // like before
  }
  // Call impossible query formatter instead of node.Format.
  sqlparser.ImpossibleQueryFormatter(buf, node)
}
```
Then, all you need to do is implement `ImpossibleQueryFormatter` in the parser.

In the case of vttablet, you can directly instantiate `sqlparser.NewTrackedBuffer(sqlparser.ImpossibleQueryFormatter)`.